lib/hydra/works/models/concerns/collection_behavior.rb

```diff
@@ -22,6 +22,39 @@ module CollectionBehavior
       type [RDFVocabularies::PCDMTerms.Collection,WorksVocabularies::WorksTerms.Collection]
       include Hydra::Works::AggregatesGenericWorks
       include Hydra::Works::AggregatesCollections
+      include Hydra::Works::BlockChildObjects
+
+      filters_association :members, as: :child_collections, condition: :works_collection?
+      filters_association :members, as: :child_generic_works, condition: :works_generic_work?
+    end
+
+    module ClassMethods
+      def indexer
+        Hydra::Works::CollectionIndexer
+      end
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Collection.
+    def works_collection?
+      true
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic Work.
+    def works_generic_work?
+      false
+    end
+
+    # @return [Boolean] whether this instance is a Hydra::Works Generic File.
+    def works_generic_file?
```

Comment:
shouldn't we treat this case just like any other object that doesn't respond to that method?  e.g. (`generic_work.child_generic_files += [Foo.new]`) Why do we have a special case for putting Collections into the generic_file association?  You should never do that, just like you should never put a `Foo` on there?